### PR TITLE
refactor: add separate OptionGroup.js

### PR DESCRIPTION
**Pull request by iambumblehead**

### CODE REVIEW

**Comment by iambumblehead** [comment]
this pipeline is terrific :)

File: src/Menu.js [hunk 1]
```diff
@@ -13,55 +14,33 @@ const Menu = ({
   noOptionsDisplay,
   className
 }) => {
-  let tabIndex = -1;
```
Comment (lwhiteley) [inline src/Menu.js]: the option tabIndex should be sequential, that is why it was defined outside the loop, 

- option 1 => tabIndex 0
- option 2 => tabIndex 1
- Group
  - option 1 => tabIndex 2
  - option 2 => tabIndex 3

Comment (iambumblehead) [reply src/Menu.js]: I see. I'll look for the best way to re-add it.

Comment (iambumblehead) [reply src/Menu.js]: feel free to add or suggest a different approach. initially I tried to make a little one-liner function, but the lint rules didn't like those and so the result is a fully defined `getTabIndexFunction` :|

File: src/Menu.js [hunk 2]
```diff
@@ -29,6 +40,7 @@ const Menu = ({
       selected={selected}
       className={className}
       onSelect={onSelect}
+      getTabIndex={getTabIndex}
```
Comment (lwhiteley) [inline src/Menu.js]: dont think this is the correct approach. should only pass a number/string. passing a function complicates the component interface. given this is not an event

Comment (iambumblehead) [reply src/Menu.js]: I have an idea let's see what you think...

Comment (iambumblehead) [reply src/Menu.js]: @lwhiteley what do you think?

**Comment by iambumblehead** [comment]
I needed to run the install again to get the husky hooks...

**Pull request merged**